eni.tf

```diff
@@ -4,7 +4,7 @@ locals {
 
 resource "aws_network_interface" "additional" {
   count           = local.additional_ips_count * var.instance_count
-  subnet_id       = var.subnet
+  subnet_id       = element(var.subnet_ids, count.index)
```

Comment:
will it work if an instance is placed in one subnets, but a network interface for the instance is placed into another?
`element` does "wrap-around"
Please review this logic.